pfcon/pfcon.py

```diff
@@ -1963,3 +1949,79 @@ def setup(self, **kwargs):
             Colors.NO_COLOUR,
             level   = 1,
             syslog  = False)
+
+    def get_pfioh_ip(self):
+        pfioh_ip = (
+            self.read_ip_by_host_name_environment_variable(
+                PFIOH_HOST_NAME_ENVIRONMENT_VARIABLE,
+                default_host_name=PFIOH_HOST_NAME_DEFAULT)
+            or self.read_from_environment(PFIOH_IP_ENVIRONMENT_VARIABLE)
+            or self.read_from_environment(HOST_IP_ENVIRONMENT_VARIABLE)
+            or read_local_host_ip()
+        )
+        self.dp.qprint(f'Found pfioh IP address: {pfioh_ip}.')
+        return pfioh_ip
+
+    def get_pman_ip(self):
+        pman_ip = (
+            self.read_ip_by_host_name_environment_variable(
+                PMAN_HOST_NAME_ENVIRONMENT_VARIABLE,
+                default_host_name=PMAN_HOST_NAME_DEFAULT)
+            or self.read_from_environment(PMAN_IP_ENVIRONMENT_VARIABLE)
+            or self.read_from_environment(HOST_IP_ENVIRONMENT_VARIABLE)
+            or read_local_host_ip()
+        )
+        self.dp.qprint(f'Found pman IP address: {pman_ip}.')
+        return pman_ip
+
+    def read_ip_by_host_name_environment_variable(
+            self, host_name_environment_variable, default_host_name=''):
+        '''For newer docker-compose'''
+
+        host_name = os.environ.get(
+            host_name_environment_variable, default_host_name
+        )
+        self.dp.qprint(f'Reading IP address of {host_name}.')
+        try:
+            ip_address = socket.gethostbyname(host_name)
+        except Exception as exception:
+            self.dp.qprint(
+                f'Could not obtain IP address of host {host_name} ({exception}).'
+            )
+            return None
+
+        if ip_address == '127.0.0.1':
+            return None
+        return ip_address
+
+
+    def read_from_environment(self, environment_variable, default=None):
+        value = os.environ.get(environment_variable, default)
+        self.dp.qprint(
+            f'Obtained {value} from environment variable {environment_variable}'
```

Comment:
f-strings are new in python3.6, so `setup.py` also needs modernizing.

example: https://github.com/FNNDSC/pfioh/blob/master/setup.py

Reply:
Github is indicating this conversation as unresolved. Could you please double-check whether this needs more work, @jennydaman?